lib/core/unittest/CLoggerTest.cc

```diff
@@ -199,6 +204,53 @@ BOOST_FIXTURE_TEST_CASE(testNonAsciiJsonLogging, CTestFixture) {
     BOOST_REQUIRE_EQUAL(messages.size(), foundMessages);
 }
 
+BOOST_FIXTURE_TEST_CASE(testWarnAndErrorThrottling, CTestFixture) {
+
+    std::ostringstream loggedData;
+    std::thread reader(makeReader(loggedData));
+
+    std::string messages[]{"Warn should only be seen once", "Error should only be seen once"};
+
+    ml::core::CLogger& logger = ml::core::CLogger::instance();
+    // logger might have been reconfigured in previous tests, so reset and reconfigure it
+    logger.reset();
+    logger.reconfigure(TEST_PIPE_NAME, "");
+
+    for (std::size_t i = 0; i < 10; ++i) {
+        LOG_WARN(<< messages[0])
+        LOG_ERROR(<< messages[1])
+    }
+
+    // reset the logger to end the stream and revert state for following tests
+    logger.reset();
+
+    reader.join();
+    std::istringstream inputStream(loggedData.str());
+    std::string line;
+    size_t foundMessages = 0;
+
+    // test that we found the messages we put in,
+    while (std::getline(inputStream, line)) {
+        if (line.empty()) {
+            continue;
+        }
+        rapidjson::Document doc;
+        doc.Parse<rapidjson::kParseDefaultFlags>(line);
+        BOOST_TEST_REQUIRE(!doc.HasParseError());
+        BOOST_TEST_REQUIRE(doc.HasMember("message"));
+        const rapidjson::Value& messageValue = doc["message"];
+        std::string messageString(messageValue.GetString(), messageValue.GetStringLength());
+        std::cout << messageString << std::endl;
```

Comment:
Should this be committed?

Reply:
No hangover.

Reply:
See [65caeb0](https://github.com/elastic/ml-cpp/pull/1615/commits/65caeb0d5cedead7d382e82993dca0959a5cfd37).